benchmarks/fp8_matmul.py

```diff
@@ -0,0 +1,276 @@
+                config.fp8_kernel,
+                config.compile,
+                f"{result.bf16_time:.4f}",
+                f"{result.fp8_time:.4f}",
+                f"{speedup:.2f}x",
+                f"{result.bf16_tflops:.2f}",
+                f"{result.fp8_tflops:.2f}",
+                f"{tflops_ratio:.2f}x",
+            ]
+        )
+    print(tabulate(rows, headers=headers, floatfmt=".4f"))
+
+    if save_path is not None:
+        with open(save_path, "w", newline="") as f:
+            writer = csv.writer(f)
+            writer.writerow(headers)
+            writer.writerows(rows)
+        print(f"💾 Results saved to: {save_path}")
+
+
+def get_configs_varying_k() -> List[ExperimentConfig]:
+    shapes = [(8192, K, 8192) for K in range(512, 8193, 512)]
+    scaling_strategies = [ScalingStrategy.PER_ROW]
+    compile_options = [False]
+    configs = []
+    fp8_kernels = [
+        FP8Kernel.SCALED_MM,
+        FP8Kernel.PERSISTENT,
+        FP8Kernel.PERSISTENT_TMA,
+        FP8Kernel.DEVICE_TMA,
+    ]
+
+    for (M, K, N), strategy, compile, kernel in itertools.product(
+        shapes, scaling_strategies, compile_options, fp8_kernels
+    ):
+        configs.append(
+            ExperimentConfig(
+                M=M, K=K, N=N, scaling_strategy=strategy, compile=compile, fp8_kernel=kernel
+            )
+        )
+    return configs
+
+
+def get_configs() -> List[ExperimentConfig]:
+    shapes = [
+        (8192, 512, 8192),
+        (1024, 20512, 2048),
+        (196608, 656, 256),
+        (1024, 1024, 137568),
+        (1024, 256, 137568),
+        (1024, 1568, 1664),
+        (1024, 1024, 43040),
+        (1024, 43040, 2048),
+        (1024, 1568, 1920),
+        (1024, 1024, 20512),
+        (137216, 1920, 384),
+        (1024, 5632, 98304),
+        (196608, 656, 624),
+        (196608, 3600, 624),
+        (1024, 1568, 640),
+        (1024, 192, 98304),
+        (1024, 1568, 1024),
+        (1024, 137568, 2048),
+        # (1024, 24576, 98304), Fails on for a large N values
+        (1024, 1024, 1024),
+    ]
+
+    scaling_strategies = [ScalingStrategy.PER_TENSOR]
+    compile_options = [False]
+    configs = []
+    fp8_kernels = [
+        FP8Kernel.SCALED_MM,
+        FP8Kernel.PERSISTENT,
+        FP8Kernel.PERSISTENT_TMA,
+        FP8Kernel.DEVICE_TMA,
+    ]
+    for (M, K, N), strategy, compile, kernel in itertools.product(
+        shapes, scaling_strategies, compile_options, fp8_kernels
+    ):
+        configs.append(
+            ExperimentConfig(
+                M=M, K=K, N=N, scaling_strategy=strategy, compile=compile, fp8_kernel=kernel
+            )
+        )
+    return configs
+
+
+def main(save_path: Optional[str] = None):
+    """Benchmark FP8 MatMul with different configurations.
+
+    Args:
+        save_path (Optional[str], optional): Path to save the results. Defaults to None.
+    """
+    torch.random.manual_seed(123)
+    configs = get_configs_varying_k()
```

Comment:
just curious, any interesting insights from varying M and N in this script?

Reply:
## M Sweeps
![image](https://github.com/user-attachments/assets/eb73ad4b-1de1-42a6-8a51-ffb6ab1861c9)
![image](https://github.com/user-attachments/assets/a30be46e-9c7b-4925-874e-54cf99d53924)![image](https://github.com/user-attachments/assets/0c5bb2aa-2fc3-422e-bea4-492d5c645270)
![image](https://github.com/user-attachments/assets/b42705ec-997a-4e73-a147-4743f61a71c8)
![image](https://github.com/user-attachments/assets/760dc760-16b5-4f2a-99d7-34e018468a9a)

## N Sweeps
Are pretty similiar
![image](https://github.com/user-attachments/assets/76f55c73-b19c-4be3-a46f-f1c1a4e4f9ba)



Soo seems like for smaller sizes scaled_mm w/ cutlass in general is much better when less compute bound and (with current non finetuned kernel config) the Triton Impl can overtake